submit/controllers/ui/new/verify_user.py

```diff
@@ -54,29 +55,22 @@ def verify(method: str, params: MultiDict, session: Session,
         'user': session.user,   # We want the most up-to-date representation.
     }
 
-    if method == 'POST':
-        if not (form.validate() and form.verify_user.data):
-            # Either the form data were invalid, or the user did not check the
-            # "verify" box.
-            raise BadRequest(response_data)
-
-        logger.debug(f'Form is valid: {form.verify_user.data}')
-
+    if method == 'POST' and form.validate() and form.verify_user.data:
         # Now that we have a submission, we can verify the user's contact
         # information. There is no need to do this more than once.
-        if not submission.submitter_contact_verified:
+        if submission.submitter_contact_verified:
+            return ready_for_next((response_data, status.OK,{}))
+        else:
             cmd = ConfirmContactInformation(creator=submitter, client=client)
-            if not validate_command(form, cmd, submission, 'verify_user'):
-                raise BadRequest(response_data)
-
-            try:
-                submission, _ = save(cmd, submission_id=submission_id)
-            except SaveError as e:
-                raise InternalServerError(response_data) from e
-            response_data['submission'] = submission
-
-        return response_data, status.SEE_OTHER, {}
-    return response_data, status.OK, {}
+            if validate_command(form, cmd, submission, 'verify_user'):
+                try:
+                    submission, _ = save(cmd, submission_id=submission_id)
+                    response_data['submission'] = submission
+                    return ready_for_next((response_data, status.OK, {}))
+                except SaveError as e:
```

Comment:
```suggestion
                except SaveError as ex:
```

Reply:
:+1:  will do this and the other exception var name changes.